cmd/coco/reconcile/reconcile.go

```diff
@@ -0,0 +1,294 @@
+	if err != nil {
+		return fmt.Errorf("failed to create a draft PR: %v", err)
+	}
+	fmt.Printf("Draft pull request #%d created: %s\n", pr.GetNumber(), pr.GetHTMLURL())
+	return nil
+}
+
+var checkMergability = func(ctx context.Context, reconcileBranchName, source, target, ownerName, repoName string, client *github.Client) (bool, error) {
+	prs, _, err := client.PullRequests.List(ctx, ownerName, repoName, nil)
+	if err != nil {
+		return false, err
+	}
+	var pr *github.PullRequest
+	for _, p := range prs {
+		if p.Head.GetRef() == source && p.Base.GetRef() == reconcileBranchName {
+			pr = p
+			break
+		}
+	}
+	if pr != nil {
+		// check if the pull request is mergable
+		if pr.GetMergeable() {
+			// perform the merge
+			commitMessage := "Merge " + reconcileBranchName + " into " + target
+			mergeRequest := &github.RepositoryMergeRequest{
+				Base:          &target,
+				Head:          &reconcileBranchName,
+				CommitMessage: &commitMessage,
+			}
+			_, _, err := client.Repositories.Merge(ctx, ownerName, repoName, mergeRequest)
+			if err != nil {
+				fmt.Println("Successfully merged reconcile branch to target branch")
+				return true, err
+			}
+			return false, err
+		} else {
+			return false, fmt.Errorf("Please re-try after resolving the merge conflicts here: %s", pr.GetURL())
+		}
+
+	} else {
+		return false, fmt.Errorf("the pull request was not found")
+	}
+}
+
+var handleTargetAhead = func(reconcileBranchName string, ownerName string, repoName string, client *github.Client, ctx context.Context) (bool, error) {
+	fmt.Print("The target branch has new commits, choose one of the following options:\n\n" +
+		"Option 1: Merge the target branch into the reconcile branch manually and rerun command `coco reconcile`\n\n" +
+		"Option 2: Automatically delete the reconcile branch and rerun the command `coco reconcile`\n\n" +
+		"Enter [1] for Option 1 or [2] for Option 2: ")
+	var input int
+	fmt.Scanln(&input)
+	switch input {
+	case 1:
+		fmt.Printf("\nPlease delete the branch %s and rerun the `coco reconcile` command", reconcileBranchName)
+	case 2:
+		return false, deleteBranch(client, ctx, ownerName, repoName, reconcileBranchName)
+	default:
+		for input != 1 && input != 2 {
+			fmt.Print("\nPlease choose either Option 1 or 2. Enter [1] for Option 1 or [2] for Option 2: ")
+			fmt.Scanln(&input)
+		}
+		if input == 1 {
+			fmt.Printf("\nPlease delete the branch %s and rerun the `coco reconcile` command", reconcileBranchName)
+		} else if input == 2 {
+			return false, deleteBranch(client, ctx, ownerName, repoName, reconcileBranchName)
+		}
+	}
+	return true, nil
+}
+
+var mergeBranches = func(ctx context.Context, client *github.Client, ownerName string, repoName string, merge *github.RepositoryMergeRequest) error {
+	_, _, err := client.Repositories.Merge(ctx, ownerName, repoName, merge)
+	return err
+}
+
+var authenticateWithGithub = func(ctx context.Context) (*github.Client, error) {
```

Comment:
I would say we put all github functionality into its own object and have a constructor `NewGithub` for it.

Reply:
this implies, that the test needs to use a constructor that returns a fake githubClient implementation